File: data structures and algorithms/java/06_021.java
public void dfsThreeWays(TreeNode node) {
   // Base case: return if node is null
   if (node == null) {
       return;
   }
   
   // Pre-order: Process BEFORE children
   System.out.println(node.value);  // Pre-order
   dfsThreeWays(node.left);
   // In-order: Process BETWEEN children
   System.out.println(node.value);  // In-order
   dfsThreeWays(node.right);
   // Post-order: Process AFTER children
   System.out.println(node.value);  // Post-order
}

File: data structures and algorithms/cs/06_021.cs
public void DfsThreeWays(TreeNode node)
{
   // Base case: return if node is null
   if (node == null)
   {
      return;
   }

   // Pre-order: Process BEFORE children
   Console.WriteLine(node.Value);  // Pre-order
   DfsThreeWays(node.Left);
   // In-order: Process BETWEEN children
   Console.WriteLine(node.Value);  // In-order
   DfsThreeWays(node.Right);
   // Post-order: Process AFTER children
   Console.WriteLine(node.Value);  // Post-order
}
